cpp/src/parquet/arrow/schema.cc

```diff
@@ -410,21 +410,66 @@ ::arrow::Result<std::shared_ptr<ArrowType>> GetTypeForNode(
   return storage_type;
 }
 
-Status NodeToSchemaField(const Node& node, int16_t max_def_level, int16_t max_rep_level,
+struct LevelInfo {
+  int16_t def_level = 0;
+  int16_t rep_level = 0;
+  int16_t repeated_ancestor_def_level = 0;
```

Comment:
Structure with these three fields seem to be in multiple numbers now? Perhaps we should simply use `LevelInfo` as a plain datatype everywhere instead of copying all three fields by hand everytime?

Reply:
consolidated into 1 in level_conversions where ina follow-up PR will use it as an argument to the levels there.